charms/knative-operator/src/charm.py

```diff
@@ -33,28 +35,70 @@ def __init__(self, *args):
 
         self._app_name = self.model.app.name
         self._namespace = self.model.name
-        self.resource_handler = KRH(
-            template_files=self._template_files,
-            context=self._context,
-            field_manager=self._namespace,
+        self._src_dir = Path("src")
+        self._template_files_ext = None
+        self._resource_handler = None
+
+        self._scraping = MetricsEndpointProvider(
+            self,
+            relation_name="metrics-endpoint",
+            jobs=[{"static_configs": [{"targets": [f"{self._otel_exporter_ip}:8889"]}]}],
         )
         self._operator_service = "/ko-app/operator"
         self._container = self.unit.get_container(self._app_name)
         for event in [self.on.install, self.on.config_changed]:
             self.framework.observe(event, self._main)
+
+        self.framework.observe(
+            self.on["otel-collector"].relation_changed, self._on_otel_collector_relation_changed
+        )
```

Comment:
Is `_changed` the appropriate event here?  `_changed` implies to me that we're expecting the other side to send data to this relation, and we want to react to that.  But I don't think that is the case here.

I think that the relation data code below is sending application level data (not data to each unit specifically), so the goal here would be "if I have any applications relating to me, start otel if it isn't started and send them the details".  For that, the `relation_created` event sounds most appropriate, as it is guaranteed to fire when an application relates to this application.

Reply:
Changed to `_created`.